Automattic/wp-calypso
# Importers: don't use importer-nux state in importers, read and drill query args instead

The importers UI has a feature where you can specify the desired importer and imported site URL in the query args:
```
https://wordpress.com/import/example.blog?engine=wix&from-site=https://example.com
```
The `/import/:site` route handler will push the `engine` and `fromSite` info to router state, and replaces the route with mere `/import/example.blog`, so that you never see the query params. But they will trigger a WIX export from the site to start immediately.

This PR removes usage of the `importer-nux` reducer when handling that feature. Instead of the page.js route handler dispatching a setter Redux action and then the `SectionImport` reading the info with Redux selectors, the route handler simply passes the info directly to the component as props.

As a result, part of the `importer-nux` reducer becomes unused and can be removed. And the `import` section now uses only the `state.imports` reducer and nothing else, making it easier to modularize.

There's also the `signup` query arg, added by the import flow in Signup: https://github.com/Automattic/wp-calypso/blob/trunk/client/signup/config/flows-pure.js#L325

The router handler was referring to it by a wrong name, `isFromSignup`, so I fixed that. But the arg is not really used: the only usage was to redirect to `/view/example.blog?welcome` rather than to `/view/example.blog` after the import is done. But the Preview code doesn't detect or handle the `welcome` arg in any way.

**How to test:**
Go to the URL mentioned at the beginning, with the `engine` and `from-site` args. Verify that the import starts right away. In case of WIX, that means that the importer reads the remote site, shows you a screenshot and asks whether it's really the site you want to import.

Alternatively, go through the `/start/import` flow. You'll be prompted to enter the site name in one of the signup steps, and then, at the end of the flow, you'll be redirected to a `/import` page where the site engine and URL will be prefilled.

@jblz worked on this feature in #27148 and related PRs. I'm not sure if the project was ever finished and if the `/start/import` flow is still active.

This is a spinoff from #48213.
